File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/03_Conditional_Statements_Advanced_Exercise/04_Fishing_Boat.js
function fishingBoat(input) {
    let budget = Number(input[0]);
    let season = input[1];
    let fishermen = Number(input[2]);

    let totalCost = 0;


    if (season == "Spring") {
        totalCost = 3000;
    } else if (season == "Summer") {  // else if (season == "Summer" || season == "Autumn")
        totalCost = 4200;
    } else if (season == "Autumn") {
        totalCost = 4200;
    } else if (season == "Winter") {
        totalCost = 2600;
    }
    
    if (fishermen <= 6) {
        totalCost *= 0.90;
    } else if (fishermen <= 11) {
        totalCost *= 0.85;
    } else if (fishermen > 11) {   // >= 12
        totalCost *= 0.75;
    }

    if (fishermen % 2 == 0 && season != "Autumn") {
        totalCost *= 0.95;
    }

    if (budget >= totalCost) {
        // let diff = budget - totalCost;
        console.log(`Yes! You have ${(budget - totalCost).toFixed(2)} leva left.`);
    } else {
        console.log(`Not enough money! You need ${(totalCost - budget).toFixed(2)} leva.`);
    }
}


// fishingBoat(["3000", "Summer", "11"]);

File: Python_Courses/Python_Basics/Python_PB_2023_10/03_Conditional_Statements_Advanced_Exercise/04_Fishing_Boat.py
budget = int(input())
season = input()  # "Spring", "Summer", "Autumn" или "Winter"
fishermen = int(input())

boat_rent = 0
discount = 0
if season == "Spring":
    boat_rent = 3000
elif season == "Summer" or season == "Autumn":
    boat_rent = 4200
elif season == "Winter":  # else:
    boat_rent = 2600

if fishermen <= 6:
    discount = 0.10
    # boat_rent *= 0.90  # boat_rent = boat_rent * (1 - 0.10) eq. boat_rent = boat_rent * 0.90
elif fishermen <= 11:  # [7; 11]
    discount = 0.15
else:  # elif fishermen >= 12:
    discount = 0.25

extra_discount = 0
# if fishermen % 2 == 0 and not season == "Autumn":  # eq "... and season != "Autumn""
if fishermen % 2 == 0 and (season == "Spring" or season == "Summer" or season == "Winter"):  # eq "... and season != "Autumn""
    extra_discount = 0.05

total_cost = boat_rent * (1 - discount) * (1 - extra_discount)  # != boat_rent * (1 - discount - extra_discount)

diff = budget - total_cost
if diff >= 0:
    print(f"Yes! You have {abs(diff) :.2f} leva left.")  # 100 => abs(100) == 100
else:
    print(f"Not enough money! You need {abs(diff) :.2f} leva.")  # eq. {-diff :.2f}
